x-pack/plugins/lens/public/xy_visualization/xy_config_panel/layer_header.tsx

```diff
@@ -59,8 +59,8 @@ function DataLayerHeader(props: VisualizationLayerWidgetProps<State>) {
   const [isPopoverOpen, setPopoverIsOpen] = useState(false);
   const { state, layerId } = props;
   const layers = state.layers.filter(isDataLayer);
-  const index = layers.findIndex((l) => l.layerId === layerId);
-  const layer = layers[index];
+  const layer = layers.find((l) => l.layerId === layerId)!;
+  const index = state.layers.findIndex((l) => l === layer)!;
```

Comment:
@dej611, what do you think about changing it to 
```js
  const index = state.layers.findIndex((l) => l.layerId === layer.layerId)!;
  ```
  
  I see the potential problem in the future if somebody will change the layer. For example:
  ```js
  l !== { ...layer }
  ```
  
  WDYT about it? Does this nit make sense?

Reply:
I can see where you are coming from, but the `layer` object comes from a `.find` local call. I'd expect if anybody would change that `find` code with a copy of it, would also change its reference checks